feature-libs/my-account/organization/components/unit/services/unit-list.service.ts

```diff
@@ -17,60 +25,118 @@ import { OrganizationTableType } from '../../shared/organization.model';
 export class UnitListService extends OrganizationListService<B2BUnit> {
   protected tableType = OrganizationTableType.UNIT;
 
+  protected initialExpanded = 1;
+
+  protected expandState = new Map<string, boolean>();
+  protected levelState = new Map<string, number>();
```

Comment:
I would not use the map to store all levels. I'd rather maintain a single value for the expanded/collapsed state, and take that into account when building the tree. I think it's easier and clear, as you could potentially rollback the levelState, and with that return to the previous expanded state.

Reply:
I hope it would be what you like: [`f2b5f8d` (#8894)](https://github.com/SAP/spartacus/pull/8894/commits/f2b5f8dc63337ee23921e5fe251d2e51c4d7316f)